lib/util/periods.js

```diff
@@ -1029,28 +1039,60 @@ shaka.util.PeriodCombiner = class {
   }
 
   /**
-   * @param {T} outputStream An audio or video output stream
-   * @param {T} candidate A candidate stream to be combined with the output
-   * @return {boolean} True if the candidate could be combined with the
-   *   output stream
+   * @param {T} a
+   * @param {T} b
+   * @return {boolean}
    *
    * @template T
    * Accepts either a StreamDB or Stream type.
    *
    * @private
    */
-  static areAVStreamsCompatible_(outputStream, candidate) {
+  static areAVStreamsExactMatch_(a, b) {
+    if (a.mimeType != b.mimeType) {
+      return false;
+    }
+    /**
+     * @param {string} codecs
+     * @return {string}
+     */
     const getCodec = (codecs) => {
       if (!shaka.util.PeriodCombiner.memoizedCodecs.has(codecs)) {
         const normalizedCodec = shaka.util.MimeUtils.getNormalizedCodec(codecs);
         shaka.util.PeriodCombiner.memoizedCodecs.set(codecs, normalizedCodec);
       }
       return shaka.util.PeriodCombiner.memoizedCodecs.get(codecs);
     };
-    // Check MIME type and codecs, which should always be the same.
-    if (candidate.mimeType != outputStream.mimeType ||
-        getCodec(candidate.codecs) != getCodec(outputStream.codecs)) {
-      return false;
+    return getCodec(a.codecs) == getCodec(b.codecs);
+  }
+
+  /**
+   * @param {T} outputStream An audio or video output stream
+   * @param {T} candidate A candidate stream to be combined with the output
+   * @return {boolean} True if the candidate could be combined with the
+   *   output stream
+   *
+   * @template T
+   * Accepts either a StreamDB or Stream type.
+   *
+   * @private
+   */
+  static areAVStreamsCompatible_(outputStream, candidate) {
+    // Check for an exact match.
+    if (!shaka.util.PeriodCombiner.areAVStreamsExactMatch_(
+        outputStream, candidate)) {
+      // It's not an exact match. See if we can do multi-codec stream instead,
+      // using SourceBuffer.changeType.
+      // TODO: add a config value for this (might require making it non-static?)
+      // TODO: also require CodecSwitchingStrategy.SMOOTH?
+      if (!shaka.media.Capabilities.isChangeTypeSupported()) {
```

Comment:
It is necessary to check the configuration because a reset in this case would produce an unwanted effect.

Reply:
Okay, I added a config value.
I'm not manually checking to see if the codec switching strategy is SMOOTH, since that's not in the config object given to the manifest parser. I just made it so that it is only on by default if smooth switching is supported, and added a warning not to use it with non-smooth switching. Would you say that's enough?